docs/config-flatcar-v1_2-exp.md

```diff
@@ -168,6 +168,8 @@ The Flatcar configuration is a YAML document conforming to the following specifi
         * **pin** (string): the clevis pin.
         * **config** (string): the clevis configuration JSON.
         * **_needs_network_** (boolean): whether or not the device requires networking.
+    * **_cex_** (object): describes the IBM Crypto Express (CEX) card configuration for the luks device.
+      * **_enabled_** (boolean): whether or not to use a CEX secure key to encrypt the luks device.
```

Comment:
If it needs changes outside of Ignition and Butane then we should reach out to them before we add it here.

Reply:
Hi @travier ,  Could you help here how can i get rid of this from flatcar?

Reply:
Probably something like: https://github.com/madhu-pillai/butane/blob/test_cex/internal/doc/butane.yaml#L49..L60

Reply:
Ah, no, looks like it's just for a substring.

Reply:
Hi @travier,
What i found is  flatcar has luks feature  too where it generates the document from the  `ignition/v2/config/doc`. The flatcar schema is configured to obtain config from `base/v0_6_exp` . For the `cex` feature i've added the `cex.enabled` in the `base/v0_6_exp`.  Adding conditions in the `butane.yaml` does not make any changes, as the docs get pulled according to the `ignition/v2/config/doc`.

Reply:
It looks like there are field filters that can be used to ensure that trying to use some base Ignition fields result in errors on some variants. See e.g. https://github.com/coreos/butane/blob/d26d80317825a24f482d9c6cca2fa80181e0082f/config/openshift/v4_17_exp/translate.go#L62-L109. I think we can do the same here.

But also, since most of the enablement work was done in Ignition and Flatcar does support Clevis at least, CEX encryption also should mostly work in Flatcar. So they might be interested in testing it as well and if it works, remove the field filters in a follow up (cc @tormath1).